### PR TITLE
Allow empty string delimiter for split function

### PR DESCRIPTION
**Pull request by joshthoward**
See [this](https://github.com/prestodb/presto/pull/12420) PR for more details. Adding for parity.

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/operator/scalar/StringFunctions.java [hunk 1]
```diff
@@ -406,6 +406,10 @@ public static Block split(@SqlType("varchar(x)") Slice string, @SqlType("varchar
             if (splitIndex < 0) {
                 break;
             }
+            if (delimiter.length() == 0) {
```
Comment (raunaqmorarka) [inline core/trino-main/src/main/java/io/trino/operator/scalar/StringFunctions.java:409]: I think we need to remove `checkCondition(delimiter.length() > 0` as well.
I hope the test is catching it

**Review by hashhar — COMMENTED** [review]
There are other split function variants which should also support this if changed here.

However I'm not convinced this is a good idea. The behaviour is different from every other database. And also the empty string doesn't occur anywhere in the string (except maybe at the beginning and end of string) so the expected output for `split('a.b.c.d', '')` should be `['a.b.c.d']`.

PostgreSQL:
```
psql> SELECT split_part('a.b.c.d', '', 1);
a.b.c.d
```

SQL Server doesn't allow empty strings as delimiters at all.

The new impl. makes sense for `regexp_split` (empty string regex matches 0 characters should it splits after every character).

**Comment by martint** [comment]
We had a long discussion about this many years ago: https://github.com/prestodb/presto/issues/5290

**Comment by joshthoward** [comment]
This popped up on an internal Jira board as something to be done. I will close and mark the issue as "Won't do"

**Pull request closed without merging**